home/screens/AccountModal/LoggedOutAccountView.tsx

```diff
@@ -98,20 +104,36 @@ export function LoggedOutAccountView({ refetch }: Props) {
 
       if (result.type === 'success') {
         const resultURL = url.parse(result.url, true);
-        const sessionSecret = resultURL.query['session_secret'] as string;
-        // usernameOrEmail is always the username https://github.com/expo/universe/blob/d3332f3b48964853191c5035fceae37aeebb1e64/server/website/scenes/_app/helpers.tsx#L119
-        const usernameOrEmail = resultURL.query['username_or_email'] as string;
-
-        if (!sessionSecret) {
+        const encodedSessionSecret = resultURL.query['session_secret'] as string;
+        if (!encodedSessionSecret) {
           throw new Error('session_secret is missing in auth redirect query');
         }
 
+        const sessionSecret = decodeURIComponent(encodedSessionSecret);
+
         dispatch(
           SessionActions.setSession({
-            sessionSecret: decodeURIComponent(sessionSecret),
+            sessionSecret,
           })
         );
-        setAccountName(usernameOrEmail);
+
+        const viewerPrimaryAccountNameResult = await ApolloClient.query<
+          Home_ViewerPrimaryAccountNameQuery,
+          Home_ViewerPrimaryAccountNameQueryVariables
+        >({
+          query: Home_ViewerPrimaryAccountNameDocument,
+          context: {
+            headers: { 'expo-session': sessionSecret },
+          },
+        });
+
+        const primaryAccountName =
+          viewerPrimaryAccountNameResult.data.meUserActor?.primaryAccount.name;
+        if (!primaryAccountName) {
+          throw new Error('Logged in user must have a primary account');
```

Comment:
in what cases would a logged in user not have a primary account?

Reply:
Robots, graphql errors during this query. We don't have error reporting / sentry for home I believe so I'm not sure what to do when an error occurs other than throw an error.